src/MailtrapWait.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Codeception\Module;
+
+/**
+ * A utility class, designed to help the user to wait until a condition turns true.
+ *
+ */
+class MailtrapWait
+{
+	/**
+	 * @var Mailtrap
+	 */
+	protected $mailtrap;
+	/**
+	 * @var int
+	 */
+	protected $timeout;
+	/**
+	 * @var int
+	 */
+	protected $interval;
+
+	public function __construct(Mailtrap $mailtrap, $timeout_in_second = null, $interval_in_millisecond = null)
+	{
+		$this->mailtrap = $mailtrap;
+		$this->timeout = isset($timeout_in_second) ? $timeout_in_second : 30;
+		$this->interval = $interval_in_millisecond ?: 250;
+	}
+
+	/**
+	 * Calls the function provided with the driver as an argument until the return value is not falsey.
+	 *
+	 * @param callable $func_or_ec
+	 * @param string $message
+	 *
+	 * @throws \Exception
+	 * @return mixed The return value of $func_or_ec
+	 */
+	public function until($func_or_ec, $message = '')
```

Comment:
@polevaultweb I think we could make this a little clearer when it comes to naming variables. Wanna give it a shot?